pkg/providers/instancetype/suite_test.go

```diff
@@ -823,6 +823,29 @@ var _ = Describe("InstanceTypes", func() {
 		ExpectScheduled(ctx, env.Client, pod)
 
 	})
+	DescribeTable(
+		"should filter out a1 instances only when the AMIFamily is AL2023",
+		func(amiFamily string) {
+			nodeClass.Spec.AMIFamily = lo.ToPtr(amiFamily)
+			nodeClass.Spec.AMISelectorTerms = append(nodeClass.Spec.AMISelectorTerms, v1beta1.AMISelectorTerm{
+				Name: "*",
+			})
+			pod := coretest.UnschedulablePod()
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass, pod)
+			instanceTypes, err := cloudProvider.GetInstanceTypes(ctx, nodePool)
+			Expect(err).To(BeNil())
+			Expect(lo.ContainsBy(instanceTypes, func(it *corecloudprovider.InstanceType) bool {
+				return lo.Contains(it.Requirements.Get(v1beta1.LabelInstanceFamily).Values(), "a1")
+			})).To(Equal(amiFamily != v1beta1.AMIFamilyAL2023))
+		},
+		Entry("AL2", v1beta1.AMIFamilyAL2),
```

Comment:
You could go either way here (I just thought of it since you have an argument here that matches the title): You can specify `nil` as the first argument and the test will just print the arguments that are using for the entry if it fails. 

Minimally, this keeps you from having to maintain consistency between the description and the arguments in this case